src/dist/replication/lib/replica_stub.cpp

```diff
@@ -1231,7 +1257,36 @@ void replica_stub::close()
         _log->close();
         _log = nullptr;
     }
+}
 
+std::string replica_stub::get_replica_dir(const char* app_type, global_partition_id gpid) const
```

Comment:
for example, you have 1.0 mapped to data dir[0] initially, but now you change the data directories and the new mapping of 1.0 points to data dir[1]. In this case, you need to move the replica 1.0 dir to dir[1] from dir[0] to ensure the deterministic mapping rule is always obeyed.